packages/file-upload/src/utils.tsx

```diff
@@ -69,24 +70,31 @@ export const getFileRejectionErrorMessage = (
 };
 
 export const getErrorSummary = (
-	rejections: FileError[],
+	rejections: RejectedFile[],
 	formattedMaxFileSize: string,
 	maxFiles: number | undefined
 ) => {
 	if (!rejections?.length) return;
 
-	const firstError = rejections[0];
+	const allErrorCodes = rejections
+		.reduce((acc, { errors }) => [...acc, ...errors], [] as FileError[])
+		.map(({ code }) => code)
+		.filter((value, index, self) => {
+			return self.indexOf(value) === index;
+		});
```

Comment:
```suggestion
	const uniqueErrorCodes = Array.from(
		new Set(
			rejections.map(({ errors }) => errors.map(({ code }) => code)).flat()
		)
	);
```

I think using `reduce` and `map` and `filter` a little confusing here. I think this may be easier to read.